### PR TITLE
exclude rather than include doctests in "run_tests.py doctest"

### PR DESCRIPTION
**Pull request by rwst**
This pull request addresses issue #1904

- [X] I hereby agree to dual licence this and any previous contributions under both
the _Biopython License Agreement_ **AND** the _BSD 3-Clause License_.

- [X] I have read the ``CONTRIBUTING.rst`` file and understand that AppVeyor and
TravisCI will be used to confirm the Biopython unit tests and ``flake8`` style
checks pass with these changes.

- [X] I have added my name to the alphabetical contributors listings in the files
``NEWS.rst`` and ``CONTRIB.rst`` as part of this pull request, am listed
already, or do not wish to be listed. (*This acknowledgement is optional.*)

### CODE REVIEW

**Comment by peterjc** [comment]
I like how this is going. I see we have some failures as follows:

```
======================================================================
ERROR: Bio.PopGen.GenePop.Utils
----------------------------------------------------------------------
Traceback (most recent call last):
  File "run_tests.py", line 508, in runTest
    optionflags=doctest.ELLIPSIS)
  File "/opt/python/2.7.14/lib/python2.7/doctest.py", line 2414, in DocTestSuite
    raise ValueError(module, "has no docstrings")
ValueError: (<module 'Bio.PopGen.GenePop.Utils' from '/home/travis/build/biopython/biopython/build/lib.linux-x86_64-2.7/Bio/PopGen/GenePop/Utils.py'>, 'has no docstrings')
======================================================================
ERROR: Bio.Restriction.Restriction_Dictionary
----------------------------------------------------------------------
Traceback (most recent call last):
  File "run_tests.py", line 508, in runTest
    optionflags=doctest.ELLIPSIS)
  File "/opt/python/2.7.14/lib/python2.7/doctest.py", line 2414, in DocTestSuite
    raise ValueError(module, "has no docstrings")
ValueError: (<module 'Bio.Restriction.Restriction_Dictionary' from '/home/travis/build/biopython/biopython/build/lib.linux-x86_64-2.7/Bio/Restriction/Restriction_Dictionary.py'>, 'has no docstrings')
----------------------------------------------------------------------
```

Those modules ought to have a minimal docstring anyway, so we should do that - which would help with resolving #1203, where I'd like to enable the ``D100`` check in ``Bio/.flake8``

For ``Bio/PopGen/GenePop/Utils.py`` this is trivial - there is an intended module docstring but it is below an import line.

In the case of ``Bio/Restriction/Restriction_Dictionary.py`` both this and the code to generate the file would need updating, see ``Scripts/Restriction/ranacompiler.py`` - We can just turn some of the ``#`` prefixed comment into a module level docstring.

**Comment by rwst** [comment]
I adapted the restriction stuff and comments match too. I left out the (C) Frederic Sohm because I'm skeptical. These are autogenerated and, if any, the (C) of the sources should be on them. Opinions?

**Comment by rwst** [comment]
Passes now. Please review. Sorry again for the number of commits.

**Comment by peterjc** [comment]
The pillow / PIL import is a puzzle, I was thinking something had changed on AppVeyor or the Python ecosystem in Jan 2019, resulting in a new failure:

e.g.

```
Python version: 2.7.15 |Anaconda, Inc.| (default, Dec 10 2018, 21:57:18) [MSC v.1500 64 bit (AMD64)]
Operating system: nt win32
===============================
ERROR: Bio.Graphics.KGML_vis
----------------------------------------------------------------------
Traceback (most recent call last):
  File "run_tests.py", line 683, in runTest
    module = __import__(name, fromlist=name.split("."))
  File "C:\projects\biopython\build\lib.win-amd64-2.7\Bio\Graphics\KGML_vis.py", line 24, in <module>
    from PIL import Image
  File "C:\Py\lib\site-packages\PIL\Image.py", line 94, in <module>
    from . import _imaging as core
ImportError: DLL load failed: The specified module could not be found.
----------------------------------------------------------------------
Ran 471 tests in 643.032 seconds
```

This worked on Monday https://ci.appveyor.com/project/biopython/biopython/builds/21768582 but failed days later with https://ci.appveyor.com/project/biopython/biopython/builds/21822450/job/xi33e20t6kyoyy3m even though the associated pull request which added  ``Bio.Graphics.KGML_vis`` to the doctests passed https://github.com/biopython/biopython/pull/1900

I found a similar issue back with Pillow 4.1 and the exact version of Python used: https://stackoverflow.com/questions/43264773/pil-dll-load-failed-specified-procedure-could-not-be-found

However, specifically installing pillow 5.3, 5.2, 5.1, 5.0, or 4.3 didn't work here (tested on my fork). However pillow 4.2 did,

https://ci.appveyor.com/project/peterjc/biopython/builds/21865494

I'm not convinced this isn't just a transient problem elsewhere... but tweaking how we test the import also seemed to work (your latest commit passed).

File: Tests/run_tests.py [hunk 1]
```diff
@@ -177,8 +177,8 @@ def is_pypy():
     EXCLUDE_DOCTEST_MODULES.append("Bio.SearchIO")
 
 try:
-    import PIL
-    del PIL
+    from PIL import Image
+    del Image
 except ImportError:
     # Missing on Jython or Python 2.4
```
Comment (peterjc) [inline Tests/run_tests.py]: This comment can be removed (not relevant to the PIL / pillow import for Image)

**Comment by rwst** [comment]
Why do I get the `flake8 BioSQL/` error now? Some change elsewhere?

**Comment by peterjc** [comment]
@rwst the flake8 error was down to a bug in a flake8 dependency, config-parser, since fixed. See https://github.com/biopython/biopython/pull/1909#issuecomment-457593325

I expect quite a few people/projects would have been caught out by this, even though it was fixed the same day - which was good.

**Comment by peterjc** [comment]
I've push some trivial additions to your branch...

**Comment by peterjc** [comment]
I've extracted the docstring work with ``git cherry-pick`` and applied that to the master for #1203 (thank you!), and will take care of the merge conflict that has made - you don't need to update this branch.

**Comment by rwst** [comment]
Looks like I can close this one.

**Comment by peterjc** [comment]
Is it too late to undelete the branch and reopen this? The core improvement has not yet been merged.

**Comment by rwst** [comment]
I was not aware that things were not merged, sorry, it seems recovered. What needs to be done?

**Comment by peterjc** [comment]
GitHub says there is a merge conflict now (not surprising as there have been a few more doctest added to ``Tests/run_tests.py``, and in ``Scripts/Restriction/ranacompiler.py`` where I did some minor style changes).

Are you familiar and confident with using ``git rebase origin/master``? If so, that would be ideal.

**Comment by rwst** [comment]
Let's see if this worked. Better no longer delay the merge if it's good.

**Comment by peterjc** [comment]
Looks like 3e65121b4a217f46e61e89e28d61346898ba3cd6 got lost, presumably during manual fixing of the merge conflicts. Leave this with me...

**Comment by peterjc** [comment]
How does this look? Rebased (again) and squashed down to one commit (still with your name).

I then included a second commit to update the Tutorial chapter.

**Comment by rwst** [comment]
Fine with me.

**Comment by peterjc** [comment]
Applied, thank you!

**Pull request merged**